rplugin/python3/magma/magmabuffer.py

```diff
@@ -264,6 +264,21 @@ def update_interface(self) -> None:
 
         self.updating_interface = False
 
+    def on_cursor_moved(self, scrolled=False) -> None:
+        selected_cell = self._get_selected_span()
+
+        if self.selected_cell == selected_cell and selected_cell is not None:
+            if selected_cell.end.lineno < self.nvim.funcs.line("w$") and self.should_open_display_window and scrolled:
```

Comment:
I don't get the purpose of this conditional, and it doesn't seem right to me.

`self.nvim.funcs.line("w$")` will give the line number of the last visible line in the window (note: this is a Vim API and line numbers are different in neovim APIs; that's why it's `<` and not `<=`, I imagine). But what if part of the cell is visible but its end is past the end of the buffer?

Furthermore, why is this done:
```python
                if self.display_window is not None: # and self.nvim.funcs.winbufnr(self.display_window) != -1:
                    self.nvim.funcs.nvim_win_close(self.display_window, True)
                    self.canvas.clear()
                    self.display_window = None
                self._show_outputs(self.outputs[selected_cell], selected_cell.end)
                self.canvas.present()
```
instead of just
```python
self.clear_interface()
self._show_outputs(self.outputs[selected_cell], selected_cell.end)
self.canvas.present()
```